repos/re-theme/src/styleInjector/injectHelpers.js

```diff
@@ -0,0 +1,222 @@
+import { isArr, isStr, exists, omitKeys, pickKeys } from '@keg-hub/jsutils'
+import { hasDomAccess } from '../helpers/hasDomAccess'
+import { addThemeEvent } from '../theme/themeEvent'
+import { Constants } from '../constants'
+
+/**
+ * Cache the current environment
+ */
+const { NODE_ENV } = process.env
+const isProduction = NODE_ENV === 'production'
+const domAccess = hasDomAccess()
+
+/**
+ * Cache holder to hold the main StyleSheet Dom element
+ */
+let KegStyleSheet
+
+/**
+ * Caches selectors already added to the Dom
+ */
+const selectorCache = new Map()
```

Comment:
I think this might be better as a `Set` type, since you're only making existence checks, right?